### PR TITLE
[core][spark] Support create database with location, comment, props using hive catalog

### LINKED ISSUE #2619: [Feature] Support create db/table using location and external table
### Search before asking

- [X] I searched in the [issues](https://github.com/apache/incubator-paimon/issues) and found nothing similar.

### Motivation

Support create db/table using location and external table

### Solution

`create db location 'xxx'` or `create table location 'xxx'` 

- For fileSystem catalog, when location is define, an exception is thrown
- For hive catalog, location sync to HMS

`create external table location 'xxx'`

- For fileSystem catalog, an exception is thrown
- For hive catalog, table_type (external) sync to HMS (spark, hive)

Other

- Drop external table, will not delete actural data
- Removed some confusing configuration

### Anything else?

_No response_

### Are you willing to submit a PR?

- [x] I'm willing to submit a PR!

### PR DESCRIPTION
**Pull request by Zouxxyy**
### Purpose

Linked issue: one step of #2619

support  create database with location, comment, props

```sql
CREATE DATABASE db LOCATION 'x' COMMENT 'x' WITH DBPROPERTIES ('k1' = 'v1', 'k2' = 'v2')
```

Note:

- for hive catalog: store all props(location, comment, other props) to hms
- for filesystem catalog: when props is not empty, do nothings but give a warn `Currently filesystem catalog can't store database properties, discard properties: {}`, because filesystem catalog can't store meta.  Besides, because db's location in filesystem catalog is stable, **when create db location 'xxx' with filesystem catalog, an exception is thrown**: `Cannot specify location for a database when using fileSystem catalog`

### CODE REVIEW

File: paimon-core/src/main/java/org/apache/paimon/catalog/Catalog.java [hunk 1]
```diff
@@ -81,7 +81,33 @@ default Optional<MetastoreClient.Factory> metastoreClientFactory(Identifier iden
      * @throws DatabaseAlreadyExistException if the given database already exists and ignoreIfExists
      *     is false
      */
-    void createDatabase(String name, boolean ignoreIfExists) throws DatabaseAlreadyExistException;
```
Comment (YannByron) [inline paimon-core/src/main/java/org/apache/paimon/catalog/Catalog.java:84]: the repeated doc is not necessary. merge them as the following one.

File: paimon-core/src/main/java/org/apache/paimon/catalog/AbstractCatalog.java [hunk 2]
```diff
@@ -105,10 +106,23 @@ public void createDatabase(String name, boolean ignoreIfExists)
...
+        if (isSystemDatabase(name)) {
+            return Collections.emptyMap();
+        }
+        if (!databaseExists(name)) {
+            throw new DatabaseNotExistException(name);
+        }
+        return loadDatabasePropertiesImpl(name);
     }
 
+    protected abstract Map<String, String> loadDatabasePropertiesImpl(String name);
```
Comment (YannByron) [inline paimon-core/src/main/java/org/apache/paimon/catalog/AbstractCatalog.java:124]: I prefer to name this method `loadDatabaseProperties0` if this is required.

**Comment by YannByron** [comment]
just leave two tiny comments.

File: paimon-core/src/main/java/org/apache/paimon/catalog/AbstractCatalog.java (hunk 2 above)
Comment (Zouxxyy) [reply paimon-core/src/main/java/org/apache/paimon/catalog/AbstractCatalog.java:124]: Just for unified code style, like createDatabase

**Comment by YannByron** [comment]
LGTM.

**Review by JingsongLi — APPROVED** [review]
+1

File: paimon-core/src/main/java/org/apache/paimon/catalog/Catalog.java [hunk 3]
```diff
@@ -71,17 +71,33 @@ default Optional<MetastoreClient.Factory> metastoreClientFactory(Identifier iden
      */
     boolean databaseExists(String databaseName);
 
+    default void createDatabase(String name, boolean ignoreIfExists)
```
Comment (JingsongLi) [inline paimon-core/src/main/java/org/apache/paimon/catalog/Catalog.java:74]: document this

**Pull request merged**